omod-1.8/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/controller/openmrs1_8/SessionController1_8.java

```diff
@@ -43,7 +49,24 @@
 	@RequestMapping(method = RequestMethod.GET)
 	@ResponseBody
 	public Object get(WebRequest request) {
-		return new SimpleObject().add("sessionId", request.getSessionId()).add("authenticated", Context.isAuthenticated());
+            boolean authenticated = Context.isAuthenticated();
+            SimpleObject session = new SimpleObject();
+            session.add("sessionId", request.getSessionId()).add("authenticated", authenticated);
+            if (authenticated) {
+                User user = Context.getAuthenticatedUser();
+                String rep = request.getParameter(RestConstants.REQUEST_PROPERTY_FOR_REPRESENTATION);
+                if(null == rep || RestConstants.REPRESENTATION_DEFAULT.equals(rep)){

```

Comment:
We never want to manually construct a simple object. We always want to return a proper representation of the User.

Pseudo code for this:

    Representation rep = getRepresentation(request); // DEFAULT if nothing is specified. And use RestService.getRepresentation(String) for this
    def userRep = ConversionUtil.convertToRepresentation(Context.getAuthenticatedUser(), rep);